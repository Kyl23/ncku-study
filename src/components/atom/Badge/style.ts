import styled from '@emotion/styled';
import ClearIcon from '@mui/icons-material/Clear';
import { color } from '~/src/theme/global';

export const BadgeStyle = styled.span`
  display: flex;
  align-items: center;
  padding: 3px 10px;
  height: 20px;

  background: ${color.backgroundYellow};
  color: ${color.darkYellow};
  font-size: 14px;
  user-select: none;

  border-radius: 30px;
`;

export const CloseIcon = styled(ClearIcon)`
  margin-left: 8px;
  font-size: 14px;
  cursor: pointer;
`;
