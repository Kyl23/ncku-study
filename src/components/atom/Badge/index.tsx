import { NextPage } from 'next';
import ClickEventListener from '../ClickEventListener';
import PropsType from '../propsType';
import { BadgeStyle, CloseIcon } from './style';

const Badge: NextPage<PropsType> = ({ styleProps, textProps, eventProps }) => {
  const { className } = styleProps;
  const { value } = textProps;
  const { onClose } = eventProps;
  return (
    <BadgeStyle className={className}>
      {value}
      {onClose && (
        <ClickEventListener>
          <CloseIcon />
        </ClickEventListener>
      )}
    </BadgeStyle>
  );
};

export default Badge;
