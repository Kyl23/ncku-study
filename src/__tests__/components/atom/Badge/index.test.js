import { fireEvent, render, screen } from '@testing-library/react';
import Badge from '~/src/components/atom/Badge';

describe('test for atom - Badge', () => {
    it('render Badge without props must be same', () => {
        const { container } = render(<Badge />);

        expect(container).toMatchSnapshot();
    });

    it('render Badge with value', () => {
        const value = 'this is a super weird test';
        const { container } = render(<Badge textProps={{ value }} />);

        expect(screen.getByText(value)).toBeTruthy();
        expect(container).toMatchSnapshot();
    });

    it('render Badge with className and value', () => {
        const className = 'testingClassName';
        const value = 'test';
        const { container } = render(
            <Badge styleProps={{ className }} textProps={{ value }} />
        );

        expect(screen.getByText(value).classList).toContain(className);
        expect(container).toMatchSnapshot();
    });

    it('render Badge with close function', () => {
        const mockClose = jest.fn();
        render(<Badge eventProps={{ onClick: mockClose }} />);

        mockClose.mockClear();
        fireEvent.click(screen.getByRole('button'));
        expect(mockClose).toBeCalledTimes(1);
    });
});
