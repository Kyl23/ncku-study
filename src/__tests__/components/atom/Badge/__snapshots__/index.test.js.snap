// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for atom - Badge render Badge with className and value 1`] = `
<div>
  <span
    class="testingClassName css-1i1ahpj"
  >
    test
  </span>
</div>
`;

exports[`test for atom - Badge render Badge with value 1`] = `
<div>
  <span
    class="css-1i1ahpj"
  >
    this is a super weird test
  </span>
</div>
`;

exports[`test for atom - Badge render Badge without props must be same 1`] = `
<div>
  <span
    class="css-1i1ahpj"
  />
</div>
`;
